examples/Example/Page/Todos/TodoCSS.hs

```diff
@@ -0,0 +1,147 @@
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Example.Page.Todos.TodoCSS (page) where
+
+import Control.Monad (forM_)
+import Data.Text qualified as T
+import Effectful
+import Example.Colors (AppColor (..))
+import Example.Effects.Todos (Todo, TodoId, Todos)
+import Example.Effects.Todos qualified as Todos
+import Example.Page.Todos.Shared
+import Example.Style qualified as Style
+import Example.View.Icon qualified as Icon
+import Example.View.Inputs (toggleCheckbox)
+import Web.Hyperbole as Hyperbole
+
+page :: (Todos :> es) => Eff es (Page '[TodosView, TodoView])
+page = do
+  todos <- Todos.loadAll
+  pure $ tag "div" id $ do
+    tag "h1" id $ text "Todos CSS"
+    tag "p" id $
+      text ("Count:" <> T.pack (show $ length todos))
+    hyper MkTodosView $ todosView FilterAll todos
+
+--- TodosView ----------------------------------------------------------------------------
+
+data TodosView = MkTodosView
+  deriving (Generic, ViewId)
+
+instance (Todos :> es) => HyperView TodosView es where
+  type Require TodosView = '[TodoView]
+
+  data Action TodosView
+    = ClearCompleted
+    | Filter FilterTodo
+    | SubmitTodo
+    | ToggleAll FilterTodo
+    | SetCompleted FilterTodo Todo Bool
+    deriving (Generic, ViewAction)
+
+  update = \case
+    SubmitTodo -> do
+      TodoForm task <- formData @(TodoForm Identity)
+      _ <- Todos.create task
+      todos <- Todos.loadAll
+      pure $ todosView FilterAll todos
+    ToggleAll filt -> do
+      todos <- filteredTodos filt
+      updated <- Todos.toggleAll todos
+      pure $ todosView filt updated
+    ClearCompleted -> do
+      todos <- Todos.clearCompleted
+      pure $ todosView FilterAll todos
+    Filter filt -> do
+      todos <- filteredTodos filt
+      pure $ todosView filt todos
+    SetCompleted filt todo completed -> do
+      _ <- Todos.setCompleted completed todo
+      todos <- filteredTodos filt
+      pure $ todosView filt todos
+   where
+    filteredTodos filt =
+      filter (isFilter filt) <$> Todos.loadAll
+
+    isFilter filt todo =
+      case filt of
+        FilterAll -> True
+        Active -> not todo.completed
+        Completed -> todo.completed
+
+todosView :: FilterTodo -> [Todo] -> View TodosView ()
+todosView filt todos = do
+  todoForm filt
+  col id $ do
+    forM_ todos $ \todo -> do
+      hyper (MkTodoView todo.id) $ todoView filt todo
+  statusBar filt todos
+
+todoForm :: FilterTodo -> View TodosView ()
+todoForm filt = do
+  let f :: TodoForm FieldName = fieldNames
+  tag "div" id $ do
+    tag "span" id $ do
```

Comment:
Is "span" necessary here? (Does the CSS specifically target span?) Why not just use `el`? 

Perhaps you are also expressing a preference to control the exact html tags used / semantic DOM?

Reply:
Yes I will definitely create nicely named functions.

By opening the PR in draft mode, my intention was to get early feedback regarding the way I managed to inject the CSS rules for this specific page, and also maybe to get advice and sharing hyperview code wich looks tricky (but you already answered that, so I'll just duplicate code for now)

So the UI is not done currently, I've only implemented the bare essentials to move forward.